sql: fix EXPLAIN ALTER  RENAME

knz opened the pull request:
Fixes #30543.

Found while investigating  #30409.

knz committed:
sql: fix EXPLAIN ALTER DATABASE RENAME

Prior to this patch EXPLAINing a database rename would actually
perform the rename (which is incorrect). This patch fixes it.

Release note (bug fix): EXPLAIN ALTER DATABASE .. RENAME .. does not
rename the database any more.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/30544)
<!-- Reviewable:end -->

knz committed:
sql: fix EXPLAIN ALTER TABLE RENAME

Prior to this patch EXPLAINing a table rename would actually
perform the rename (which is incorrect). This patch fixes it.

Release note (bug fix): EXPLAIN ALTER TABLE .. RENAME .. does not
rename the database any more.

knz committed:
sql: fix EXPLAIN ALTER TABLE RENAME COLUMN

Prior to this patch EXPLAINing a column rename would actually
perform the rename (which is incorrect). This patch fixes it.

Release note (bug fix): EXPLAIN ALTER TABLE .. RENAME COLUMN .. does
not rename the column any more.

knz committed:
sql: fix EXPLAIN ALTER INDEX RENAME

Prior to this patch EXPLAINing an index rename would actually
perform the rename (which is incorrect). This patch fixes it.

Release note (bug fix): EXPLAIN ALTER INDEX .. RENAME .. does
not rename the index any more.

knz commented:
thanks

bors r+

RaduBerinde commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>





<!-- Sent from Reviewable.io -->
